editor/plugins/gpu_particles_2d_editor_plugin.cpp

```diff
@@ -57,6 +57,29 @@ void GPUParticles2DEditorPlugin::_file_selected(const String &p_file) {
 	emission_mask->popup_centered();
 }
 
+void GPUParticles2DEditorPlugin::_selection_changed() {
+	List<Node *> selected_nodes = editor->get_editor_selection()->get_selected_node_list();
+
+	if (selected_particles.is_empty() && selected_nodes.is_empty()) {
+		return;
+	}
+
+	for (GPUParticles2D *SP : selected_particles) {
+		SP->set_show_visibility_rect(false);
+	}
+	selected_particles.clear();
+
+	if (!selected_nodes.is_empty()) {
```

Comment:
This condition is not needed. If the list is empty, the iteration will just end immediately (not to mention it's the same thing you are checking at the beginning).

Reply:
fixed, thanks a lot :)

Reply:
I don't see anything changed here though...

Reply:
Hm, weird. Something went wrong there. Now it should be fine.
https://github.com/godotengine/godot/pull/55052/files#diff-d8d4979351e6c6b725e0e1d5ba45a028f4f8ff539bce4f4768b66a45266cac30R72